rust: Provide async API

cathay4t opened the pull request:
Introducing rust ASYNC API:

 * `NetworkState::retrieve_async()`
 * `NetworkState::apply_async()`

Changed nispor plugin code to use async API.

Existing test cases is enough to test this patch.

Considering nispor need tokio, we are not introducing new dependency.

Benefits:
 * Allowing nmstate rust crate user to use their own async runtime
   instead of tokio.
 * Replace ctrlc with `tokio` signal feature which is ASYNC only.
 * Upgrade zbus 1.x to zbus 4.x which is ASYNC only.

liangwen12year commented:
Nit, can you provide the reasons for introducing async API in the commit message ?

cathay4t committed:
rust: Provide async API

Introducing rust ASYNC API:

 * `NetworkState::retrieve_async()`
 * `NetworkState::apply_async()`

Changed nispor plugin code to use async API.

Existing test cases is enough to test this patch.

Considering nispor need tokio, we are not introducing new dependency.

Benefits:
 * Allowing nmstate rust crate user to use their own async runtime
   instead of tokio.
 * Replace ctrlc with `tokio` signal feature which is ASYNC only.
 * Upgrade zbus 1.x to zbus 4.x which is ASYNC only.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Benefits included in commit message and PR comment.
